internal/shim-sgx/src/handler.rs

```diff
@@ -393,10 +432,26 @@ impl<'a> SyscallHandler for Handler<'a> {
                         let req = request!(SYS_GETATT => 0, 0, shim_buf_ptr.as_ptr(), buf_len);
                         let result = unsafe { self.proxy(req)? };
 
-                        // TODO: Call Report
-                        // See https://github.com/enarx/enarx-keepldr/issues/92
+                        // Read TargetInfo data out of the Block
+                        let ti = [0u8; 512];
+                        let c = self.new_cursor();
+                        c.copy_into_slice(buf_len, b.as_mut(), buf_len)
+                            .or(Err(libc::EFAULT))?;
+
+                        // Create a TargetInfo type from TargetInfo data
+                        let mut target_info: TargetInfo = Default::default();
+                        let f = read_u64(&ti[32..40], ByteOrder::LittleEndian).unwrap();
+                        let x = read_u64(&ti[40..48], ByteOrder::LittleEndian).unwrap();
+                        let flags = Flags::from_bits(f).unwrap();
+                        let xfrm = Xfrm::from_bits(x).unwrap();
+                        let att = Attributes::new(flags, xfrm);
+                        target_info.mrenclave.copy_from_slice(&ti[0..32]);
+                        target_info.attributes = att;
+
+                        let data = ReportData([0u8; 64]);
+                        let _report: Report = unsafe { target_info.get_report(&data) };
```

Comment:
I thought we agreed on not parsing the get_attestation data in the shims, but only pass them through as byte buffers?

Reply:
Hmm, maybe that should be revised. The Report has to be generated in the shim, because it can only be generated inside an enclave.

Reply:
ah, oh, ok

@npmccallum your thoughts?